scipy/stats/_unuran/unuran_wrapper.pyx.templ

```diff
@@ -0,0 +1,1075 @@
+# cython: language_level=3
+
+
+# Expression below is replaced by ``DEF NPY_OLD = True`` for NumPy < 1.19
+# and ``DEF NPY_OLD = False`` for NumPy >= 1.19.
+DEF NPY_OLD = isNPY_OLD
+
+
+cimport cython
+from cpython.object cimport PyObject
+cimport numpy as np
+IF not NPY_OLD:
+    from cpython.pycapsule cimport PyCapsule_IsValid, PyCapsule_GetPointer
+    from numpy.random cimport bitgen_t
+from scipy._lib.ccallback cimport ccallback_t
+from scipy._lib.messagestream cimport MessageStream
+from .unuran cimport *
+import threading
+import numpy as np
+from scipy.stats._distn_infrastructure import argsreduce
+
+
+__all__ = ['UNURANError', 'TransformedDensityRejection', 'DiscreteAliasUrn']
+
+
+cdef extern from "Python.h":
+    PyObject *PyErr_Occurred() nogil
+
+
+# Internal API for handling Python callbacks.
+# TODO: Maybe, support ``LowLevelCallable``s in the future?
+cdef extern from "unuran_callback.h":
+    int init_unuran_callback(ccallback_t *callback, fcn, extra_args) except -1
+    int release_unuran_callback(ccallback_t *callback) except -1
+
+    double pdf_thunk(double x, const unur_distr *distr) nogil
+    double dpdf_thunk(double x, const unur_distr *distr) nogil
+    double cont_cdf_thunk(double x, const unur_distr *distr) nogil
+    double pmf_thunk(int x, const unur_distr *distr) nogil
+    double discr_cdf_thunk(int x, const unur_distr *distr) nogil
+
+    void error_handler(const char *objid, const char *file,
+                       int line, const char *errortype,
+                       int unur_errno, const char *reason) nogil
+
+
+class UNURANError(RuntimeError):
+    """Raised when an error occurs in the UNU.RAN library."""
+    pass
+
+
+IF not NPY_OLD:
+    cdef object get_numpy_rng(object seed = None):
```

Comment:
We have some custom function for that in QMC. We should merge them in util IMO.

Reply:
I looked at those and I don't get why both the `_lib` and `qmc` version of `check_random_state` don't support `SeedSequence`. I have added support for that here, hence, introducing yet another version of `check_random_state`. Is there any reason we should not allow users to pass `SeedSequence`? If not, I guess we can let this be for now and unify all the versions of `check_random_state` in another PR.

Reply:
There is no reason AFAIK. So I am +1 for introducing `SeedSequence` in any/both of the existing ones.

Reply:
I guess we should tackle this in a different PR then. Might help better document the introduction of `SeedSequence` that way.